server/src/main/java/org/apache/druid/discovery/DruidLeaderClient.java

```diff
@@ -256,15 +257,21 @@ public String findCurrentLeader()
     );
   }
 
-  private String getCurrentKnownLeader(final boolean cached) throws IOException
+  private String getCurrentKnownLeader(final boolean cached)
   {
     final String leader = currentKnownLeader.accumulateAndGet(
         null,
         (current, given) -> current == null || !cached ? pickOneHost() : current
     );
 
     if (leader == null) {
-      throw new IOE("No known server");
+      throw DruidException.forPersona(DruidException.Persona.ADMIN)
+                          .ofCategory(DruidException.Category.NOT_FOUND)
```

Comment:
it should be `RUNTIME_FAILURE` and not `NOT_FOUND`.

Reply:
I wasn't sure between the two. I also think `RUNTIME_FAILURE` is apt since the leader server wasn't set and hence couldn't be found due to some failure.